rpc/debug/trace.go

```diff
@@ -0,0 +1,115 @@
+// +build evm
+
+package debug
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/eth/ethapi"
+	"github.com/ethereum/go-ethereum/eth/tracers"
+	"github.com/pkg/errors"
+	abci "github.com/tendermint/tendermint/abci/types"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	ttypes "github.com/tendermint/tendermint/types"
+
+	"github.com/loomnetwork/loomchain"
+	"github.com/loomnetwork/loomchain/store"
+)
+
+func TraceTransaction(
+	app loomchain.Application,
+	blockstore store.BlockStore,
+	startBlockNumber, targetBlockNumber, txIndex int64,
+	config eth.TraceConfig,
+) (trace interface{}, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = errors.Errorf("loomchain panicked %v", r)
+		}
+	}()
+
+	if err := runUpTo(&app, blockstore, startBlockNumber, targetBlockNumber, txIndex); err != nil {
+		return nil, err
+	}
+
+	block, err := blockstore.GetBlockByHeight(&targetBlockNumber)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting block information at height %v", targetBlockNumber)
+	}
+
+	tracer, err := createTracer(config)
+	if err != nil {
+		return nil, err
+	}
+	if err := app.SetTracer(tracer, false); err != nil {
+		return nil, err
+	}
+
+	result := app.DeliverTx(block.Block.Data.Txs[txIndex])
+
+	switch tracer := tracer.(type) {
+	case *vm.StructLogger:
+		return &ethapi.ExecutionResult{
+			Gas:         5,
+			Failed:      result.Code != abci.CodeTypeOK,
+			ReturnValue: fmt.Sprintf("%x", result),
```

Comment:
Is `fmt.Sprintf` sufficient or is the resulting hex string required to have an even number of digits?

Reply:
Inspired by https://github.com/loomnetwork/go-ethereum/blob/loomchain/eth/api_tracer.go#L613 I am assuming it just all works?!